src/github.com/couchbase/sync_gateway/rest/server_context.go

```diff
@@ -724,12 +724,22 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(config *DbConfig, useExisti
 		revCacheSize = db.KDefaultRevisionCacheCapacity
 	}
 
+	unsupportedOptions := &db.UnsupportedOptions{}
+	if config.Unsupported != nil {
+		if config.Unsupported.UserViews != nil {
+			if config.Unsupported.UserViews.Enabled != nil {
+				unsupportedOptions.EnableUserViews = *config.Unsupported.UserViews.Enabled
+			}
+		}
+	}
+
```

Comment:
Gotcha, yeah the 1. case is discussed in [this SO post](http://stackoverflow.com/questions/18088294/how-to-not-marshal-an-empty-struct-into-json-with-go) and seems to be the accepted way of dealing with that.